client/src/main/java/org/eclipse/hono/client/impl/AbstractRequestResponseClient.java

```diff
@@ -510,13 +531,31 @@ private R getRequestResponseResult(final Message message) {
      *                  (see {@link AbstractHonoClient#setApplicationProperties(Message, Map)}
      */
     private Message createMessage(final String subject, final Map<String, Object> appProperties) {
+        return createMessage(subject, getDefaultMessageTargetAddress(), appProperties);
+    }
+
+    /**
+     * Build a Proton message with a provided subject (serving as the operation that shall be invoked).
+     * The message can be extended by arbitrary application properties passed in.
+     *
+     * @param subject The subject system property of the message.
+     * @param requestTargetAddress The target address of the message.
+     * @param appProperties The map containing arbitrary application properties.
+     *                      Maybe null if no application properties are needed.
+     * @return The Proton message constructed from the provided parameters.
+     * @throws NullPointerException if the subject is {@code null}.
+     * @throws IllegalArgumentException if the application properties contain not AMQP 1.0 compatible values
+     *                  (see {@link AbstractHonoClient#setApplicationProperties(Message, Map)}
+     */
+    private Message createMessage(final String subject, final String requestTargetAddress,
+            final Map<String, Object> appProperties) {
 
         Objects.requireNonNull(subject);
         final Message msg = ProtonHelper.message();
         final String messageId = createMessageId();
         AbstractHonoClient.setApplicationProperties(msg, appProperties);
-        if (!targetAddress.equals(getLinkTargetAddress())) {
-            msg.setAddress(targetAddress);
+        if (!requestTargetAddress.equals(linkTargetAddress)) {
+            msg.setAddress(requestTargetAddress);
```

Comment:
is there a problem with *always* setting the message's address to the given requestTargetAddress?

Reply:
I don't think so. I'll remove the condition.